resources/js/composables/useEvents.ts

```diff
@@ -0,0 +1,42 @@
+import { ref, watch } from 'vue'
+import api from '@/services/api'
+import type { RawEvent } from '@/types/events'
+
+export function useEvents() {
+  const events = ref<RawEvent[]>([])
+  const search = ref<string>('')
```

Comment:
You can also modify the type of the `search` variable.

```suggestion
  const search = ref<string | null>(null)
```